scripts/v6/05-asset-scoa.sql

```diff
@@ -143,3 +143,21 @@ CREATE TABLE [SCOADepreciationStatus] (
 	[Information] TEXT NULL,
 	CONSTRAINT [PK_SCOADepreciationStatus] PRIMARY KEY (rowID)
 ) ON [PRIMARY];
+
+EXECUTE('CREATE FUNCTION getSolarFinancialPeriod(@imqsFinPeriod VARCHAR(6)) RETURNS VARCHAR(6) AS
+BEGIN
+	DECLARE @PERIODS_IN_YEAR INT = 12;
+	DECLARE @ROLLOVER_PERIOD INT = 7;
+
+	DECLARE @imqsYear INT = CAST(SUBSTRING(@imqsFinPeriod, 2, 3) AS INT);
+	DECLARE @imqsPeriod INT = CAST(SUBSTRING(@imqsFinPeriod, 5, 2) AS INT);
+	DECLARE @solarYear1 VARCHAR(2) = CAST(@imqsYear-1 AS VARCHAR(2));
+	DECLARE @solarYear2 VARCHAR(2) = CAST(@imqsYear AS VARCHAR(2));
+	DECLARE @diff INT = @ROLLOVER_PERIOD - @imqsPeriod;
+	DECLARE @solarYear VARCHAR(4) = SUBSTRING(@imqsFinPeriod, 0, 3) + CASE WHEN @diff > 0 THEN @solarYear1 ELSE @solarYear2 END;
```

Comment:
I'm confused by : `SUBSTRING(@imqsFinPeriod, 0, 3)`
Shouldn't it be  `SUBSTRING(@imqsFinPeriod, 1, 2)`?

Reply:
Java style :-)

Reply:
![image](https://user-images.githubusercontent.com/4902911/32936522-42adccde-cb7d-11e7-906b-f81aa234a0c7.png)
They actually produce the same result (because no additional chars are found at position 0), but you guys are 100% correct, sql server is 1-based, so I'll make the fix.